proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,540 @@
+  | 'increasing';
+
+type ChannelValue = string | number | null;
+```
+
+### New Color Functions
+
+```ts
+export class SassColor extends Value {
+```
+
+#### `space`
+
+Returns the name of [`internal`]'s space.
+
+```ts
+get space(): KnownColorSpace;
+```
+
+#### `channels`
+
+Returns an array of channel values for [`internal`], with missing channels
+converted to `0`.
+
+```ts
+get channels(): string | number;
+```
+
+#### `channelsOrNull`
+
+Returns an array of channel values for [`internal`], with missing channels
+converted to `null`.
+
+```ts
+get channelsOrNull(): ChannelValue;
+```
+
+#### `isLegacy`
+
+Returns whether [`internal`] is in a legacy color space (`rgb`, `hsl`, or
+`hwb`).
+
+```ts
+get isLegacy(): boolean;
+```
+
+#### `isInGamut`
+
+Returns whether [`internal`] is in-gamut for its color space (as opposed to
+having one or more of its channels out of bounds, like `rgb(300 0 0)`).
+
+```ts
+get isInGamut(): boolean;
+```
+
+#### `channel`
+
+Returns the value of the given `channel` in [`internal`], after converting it to
+`space` if necessary. It should be used instead of the old channel-specific
+functions such as `color.red()` and `color.hue()`.
+
+```ts
+channel(options: {
+  channel: ChannelName;
+  space?: KnownColorSpace;
+}): ChannelValue;
+```
+
+#### `isChannelMissing`
+
+Returns whether the given `channel` of [`internal`] is missing. Missing channels
+can be explicitly specified using the special value `none` and can appear
+automatically when [toSpace()] returns a color with a powerless channel.
+
+```ts
+isChannelMissing(options: {channel: ChannelName}): boolean;
+```
+
+[toSpace()]: #tospace
+
+#### `isChannelPowerless`
+
+Returns whether the given `channel` of [`internal`] is powerless in `space`,
+defaulting to its own color space. A channel is "powerless" if its value doesn't
+affect the way the color is displayed, such as hue for a color with 0 chroma.
+Throws an error if `channel` is not a channel in `space`.
+
+```ts
+isChannelPowerless(options: {
+  channel: ChannelName;
+  space?: KnownColorSpace;
+}): boolean;
+```
+
+#### `toSpace`
+
+Returns the result of converting [`internal`] to `space` as a new SassColor.
+
+```ts
+toSpace(space: KnownColorSpace): SassColor;
+```
+
+#### `toGamut`
+
+Returns [`internal`] constrained to its space's gamut as a new SassColor. This
+is generally not recommended since even older browsers will display out-of-gamut
+colors as best they can, but it may be necessary in some cases.
+
+```ts
+toGamut(): SassColor;
+```
+
+#### `changeChannels`
+
+Returns a new SassColor as the result of changing some of [`internal`]'s
+channels. The `space` value defaults to the `space` of [`internal`], and any
+combination of channels in that space may be changed. Throws an error if any
+`channel` is not present in `space`.
+
+* If `space` is not defined, let `space` be the value of [`space(internal)`].
+
+* Let `arguments` be the key value pairs in `channels` as keyword arguments.
```

Comment:
"as keyword arguments" isn't clear here, because "keyword arguments" aren't a Sass value type. If you're passing this to a Sass rest argument, it should be a map, but then you'd also need to be explicit about how to convert from JS Strings, Numbers, and `undefined`s to Sass values.

Reply:
I ended up replacing this with an alternate method that doesn't pass it to a Sass rest argument.